### ISSUE TITLE
Link order differs between admin bar and admin overview pages

### ISSUE THREAD

**Issue by BWPanda** [backdrop/backdrop-issues#4885]
The order of links sometimes differs between the admin bar, and the admin overview pages. See the screenshot below for an example:

![image](https://user-images.githubusercontent.com/2385329/104858718-670ec900-5975-11eb-9286-4e49ba9a3be1.png)

I think we should standardise the order of these links. I believe (but am not 100% sure) that the order of links in the admin bar is set specifically based around what links users will mostly use (and they're put at the top of their respective menus). If that is indeed the case, we should use this same ordering for the links on the admin overview pages.

**Comment by klonos** [comment]
> I believe (but am not 100% sure) that the order of links in the admin bar is set specifically based around what links users will mostly use (and they're put at the top of their respective menus).

Yup 👍 ...basically it is alphabetically by default, with the weight of the most-commonly used ones adjusted/hard-coded to "bubble up" or be placed in a specific position.

I agree that the order the menus are shown on the admin menu should be reflected in the admin overview pages.

**Comment by olafgrabienski** [comment]
> the order the menus are shown on the admin menu should be reflected in the admin overview pages

Makes sense but not ordering alphabetically on a page feels also strange. Maybe it would help to style the region with the 'featured' items slighty differently, or to add some kind of separator between featured and regular items.

**Comment by indigoxela** [comment]
Re the alphabetical order of sub-items in the admin menu as long as all have a weight of "0"... This has annoyed me several times when switching the interface language. I have to remember, where the items I'm searching for appear in that language - they feel like jumping around ... Until I set a weight for them - then they "settle down". :wink:

There's probably nothing we can do, though. Menu item weights are configurable.

**Comment by BWPanda** [comment]
Here's a very simple PR: https://github.com/backdrop/backdrop/pull/4022

It seems that only the Seven theme shows links in a different order than the Admin bar; other themes are fine. This is because Seven overrides the theme function for displaying admin block content, and does its own sorting there. This is a result of: https://github.com/backdrop/backdrop-issues/issues/2828 I'm not sure if we should just revert the sorting change, or look for another alternative solution for whatever the problem was back then (if we do so, I'd recommend fixing the problem for *all* themes, not just Seven). Pinging @jenlampton, @klonos, @olafgrabienski and @quicksketch to get their feedback here since they all contributed to the previous issue.

I've tested the following pages:
- /admin
- /admin/index
- /admin/config
- /admin/config/system

The only regression I've noticed is on `/admin/index` where the '* permissions' links have moved from the top of each block to the bottom. Other than that, all links now appear in the order as per the admin bar.

**Comment by jenlampton** [comment]
The current PR just un-does the progress made in the other issue.

This was from the other issue:

> But if we rearrange the menu to be alphabetical by default, that would make them consistent at least initially.

**Comment by BWPanda** [comment]
Depends on your definition of 'consistent'...

You can either make the page links alphabetical, in which case they're consistent with other page links, but inconsistent with the same links in the admin bar; or you can make the page links match the order of the admin bar links (for consistency) and then all page links are also consistent with each other (they're all ordered according to the admin bar).

I find ordering the same links in one place (pages) one way, but then another way elsewhere (admin bar) to be *inconsistent*. Also, the fact that one theme displays links one way, and another theme another way, is also inconsistent.

So I think 'consistency' as an argument for the previous issue doesn't work...
